### ISSUE TITLE
make process names of capability providers friendlier for sysadmins

### ISSUE THREAD

**Issue by stevelr** [wasmCloud/wasmCloud#237]
Capability providers run as an executable whose name is derived from the machine architecture and os, such as "x86_64-linux", which is the name given when the par file is created. One problem with this is that all provider processes would appear to have the same in tools such as `ps`, `top`, `netdata`, default grafana dashboards, etc.

If one process is problematic, such as using too much memory or cpu, a sysadmin needs to be able to identify which one.

If the sysadmin were to view the full path to the executable, (which can be done with different ps flags) they can get the provider's public key, which can be looked up to figure out the provider name, but it would be much more intuitive to have `top` display "Httpserver". The link name should be included in the display name, to help the sysadmin distinguish between instances of the same provider.

Some ideas for resolving:
- the host could save the exe as a different file name, composed of a server name part plus a link name ("Httpserver_default"). The name part could come from the `--name` parameter that was passed to the `wash par create` command. 
- Alternately, the provider can change its own process name on startup, using a system call. This is a stable and common operation, but it's specific to linux. The process name is limited to 15 visible characters (16 with null terminator), so it might be truncated, as in `httpserver_defa`. 

Because of the character limitation, if the file rename solution is used, the host should only use the first 10 characters of the `--name` parameter, to make sure there is room to display a portion of the link name.

@autodidaddict thoughts?  Adding the change to provider_main would be straightforward, but it would be limited to linux, and only automatic if people use our libraries to implement their provider.

**Comment by autodidaddict** [comment]
Why not use the instance ID? This is information passed to the provider and managed by the host. That would let you correlate metrics and events with the process running on the machine.

**Comment by stevelr** [comment]
For metrics and events, we should definitely be logging instance id.
If the installation has a metrics server, an operator should be able to figure out all the data they need. If we log process id with the instance id, then the operator can also go back to correlate it with what's in top.

If you meant that the process name could be changed to the instance id _also_ (or at least its first 15 chars) that would be useful to an operator if wash or the web ui has an easy way to lookup an instance-id.

I might want to dedicate one or two prefix characters of the process name, like 'V' or 'WC' so that it doesn't look completely random, and it would be easy to filter on in tools like top or grafana.

**Comment by autodidaddict** [comment]
How about naming the executable something like `cap-[instance id]` ?

**Comment by stevelr** [comment]
sounds good to me

**Comment by autodidaddict** [comment]
Implementing this and noticed that there's an unintended consequence - by naming the extracted bytes the same as the instance ID, we have two issues:
* the bytes from the PAR will _never_ be cached, we will always extract from the PAR now because no filename duplication happens
* because no filename duplication happens, we now leave `cap-{guid}` files lying all over the `/tmp/wasmcloudcache/Vxxx` directories, which means we now have to clean that up ourselves.

**Comment by autodidaddict** [comment]
Come to think of it, I'm not even convinced that we need to expose the executable filename in the `all_providers()` function. It can be something left for developers (and possibly washboard) to use when examining a single `ProviderModule` pid.

**Comment by autodidaddict** [comment]
> to have top display "Httpserver". 

In a production environment, no one is going to be inside a box (container) running `top` on that machine. Some central authority will alert that a process on a managed machine/container is running away and consuming too many resources, and that authority may even automatically kill/restart that process. 

The more I look at the consequences of using instance IDs in the executable filenames the more I dislike the price to be paid for it.

We could use a filename like `wasmcloud_httpserver_default` which is just a normalization of the contract ID and link name combo.

**Comment by brooksmtownsend** [comment]
It sounds to me like storing the executable path would help tools that support interrogating the runtime (anything with control interface access) but tools like `top` will still report the `x86-linux` name as the executable. I'm good with adding the executable path for now and then modifying the naming scheme in the future if the need arises (as it won't be a breaking changes).

edit because of your most recent comment: I like `wasmcloud_httpserver_default`. That wouldn't result in any caching changes as we have both of those pieces of information when we start the provider, so we could lookup the filename right?

**Comment by autodidaddict** [comment]
Right, renaming to `wasmcloud_httpserver_default` wouldn't break the existing caching mechanism, because it's still unique to that particular provider (this executable is stored in `/tmp/wasmcloudcache/Vxxx/{revision}`

**Comment by brooksmtownsend** [comment]
I considered suggesting cutting off the `wasmcloud_` prefix, but that would mean that the executable implementing `evilcorp:httpserver` would have the same executable name and could cause unnecessary confusion
